core/include/kernel/thread.h

```diff
@@ -244,7 +244,8 @@ bool thread_is_in_normal_mode(void);
 bool thread_is_from_abort_mode(void);
 
 /**
- * Allocates data for payload buffers.
+ * Allocates data for payload buffers only shared with user space applications. 
+ * Ensure consistency with the enumeration `THREAD_SHM_TYPE_APPLICATION`.
```

Comment:
That not entirely true: `only` is misleading. Maybe prefer `Allocate data for payload buffers shared with a non-secure user space application.`

Prefer drop the back quotes around `THREAD_SHM_TYPE_APPLICATION`.

Reply:
fixed